### ISSUE TITLE
Reactive Programming

### ISSUE THREAD

**Issue by rjmk** [dwyl/tudo#88]
We are being held up by processes that are returning a lot of data, just not _all at once_ (e.g. `getIssues`). Would it be worth looking into reactive programming to give the user the data we get as we get it?

**Comment by nelsonic** [comment]
@rjmk do you _mean_ streaming the data to the client using WebSockets...? :wink:

**Comment by NataliaLKB** [comment]
Last week at the Guardian there was a tech talk regarding Reactive Programming from @theefer.

For one of his talks he prepared [this dojo](https://github.com/theefer/funjs-reactive-loop). I haven't done it yet... (on my list of learning material) but it does look good, and may be worth taking a peak at @rjmk !

**Comment by nelsonic** [comment]
What? a @TheGuardian chap _not_ doing Scala/Akka/Play? A demo written in Node.js? what? :open_mouth: 
(Looks good. But I still don't _understand_ how this is _different_ from the "old" way of _streaming data_ to the client ... is "_Reactive_" the new WebSockets?)

**Comment by rjmk** [comment]
@NataliaLKB Thanks that looks awesome! :+1:

@nelsonic I think probably websockets should be fine for us because the main differentiating feature of reactive programming (as I understand it (not very much)) is the in composing data and events from several sources. A properly extended stream can certainly fulfil the observer role that reactive programming seems to expect.

**Comment by Jbarget** [comment]
The new plan involves initialising a postgres database with all the relevant issue data as a one off operation. All requests from a user for issues will be to our db rather than github.

To keep our database up to date we're going to use github repository webhooks listening out for issue changes across all repos and organisation webhooks for when a new repo is created

https://github.com/dwyl/tudo/issues/179
https://github.com/dwyl/tudo/issues/180

if the above sounds good am I ok to close this issue?

**Comment by iteles** [comment]
We can close this issue for now and reopen if it becomes relevant again.
